Dart/src/com/jetbrains/lang/dart/ide/runner/server/vmService/DartVmServiceDebugProcess.java

```diff
@@ -279,13 +286,13 @@ public void guessRemoteProjectRoot(@NotNull final ElementList<LibraryRef> librar
 
         LOG.assertTrue(file.getPath().startsWith(localProjectRoot.getPath() + "/"), file.getPath() + "," + localProjectRoot.getPath());
         final String relPath = file.getPath().substring(localProjectRoot.getPath().length()); // starts with slash
-        if (!relPath.startsWith("/lib/") && remoteUri.endsWith(relPath)) {
+        if (relPath.startsWith("/lib/") && remoteUri.endsWith(relPath)) {
           howManyFilesMatch++;
           myRemoteProjectRootUri = remoteUri.substring(0, remoteUri.length() - relPath.length());
         }
       }
 
-      if (howManyFilesMatch == 1) {
+      if (howManyFilesMatch > 0) {
```

Comment:
> I worry that we'll reject a match and come up with no value for myRemoteProjectRootUri

We always assign myRemoteProjectRootUri to something and here we decide whether to continue heuristics. If we are not sure our guess is the best one (howManyFilesMatch > 1) then we try to guess better with another file name.